lib/external/math.dart

```diff
@@ -40,4 +45,98 @@ ASTNode funcRand(Runtime runtime, ASTNode self, List<ASTNode> args) {
+      } break;
+    case ASTType.AST_DOUBLE: 
+      switch(x.type) {
+        case ASTType.AST_INT: return DoubleNode()..doubleVal = atan2(y.doubleVal, x.intVal);
+        case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = atan2(y.doubleVal, x.doubleVal);
+        default:
+      } break;
+    default:
+  }
+  throw const UnexpectedTypeException('The atan2 method only takes [double, int] argument types for argument 1 and 2');
+}
+
+ASTNode Function(Runtime, ASTNode, List<ASTNode>) funcTrigo(String name){
+  final ASTNode Function(Runtime, ASTNode, List<ASTNode>) Function(ASTNode Function(ASTNode)) func = (Function(ASTNode) callback) {
+    return (Runtime runtime, ASTNode self, List<ASTNode> args) {
+      runtimeExpectArgs(args, [ASTType.AST_ANY]);
+      final radian = args[0];
+      switch(radian.type){
+        case ASTType.AST_INT: 
+        case ASTType.AST_DOUBLE: 
+          return callback(radian);
+        default:
+      }
+      throw UnexpectedTypeException('The $name method only accept [double, int] argument types');
+    };
+  };
+  switch(name){
+    case 'sin': return func(funcSin);
+    case 'cos': return func(funcCos);
+    case 'tan': return func(funcTan);
+    case 'asin': return func(funcASin);
+    case 'acos': return func(funcACos);
+    case 'atan': return func(funcATan);
+  }
+  throw Exception('Undefined trigo function $name');
+}
+
+ASTNode funcSin(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = sin(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = sin(radian.doubleVal);
+    default: return NullNode();
+  }
+}
+
+ASTNode funcCos(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = cos(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = cos(radian.doubleVal);
+    default: return NullNode();
+  }
+}
+
+ASTNode funcTan(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = tan(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = tan(radian.doubleVal);
+    default: return NullNode();
+  }
+}
+
+ASTNode funcASin(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = asin(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = asin(radian.doubleVal);
+    default: return NullNode();
+  }
+}
+
+ASTNode funcACos(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = acos(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = acos(radian.doubleVal);
+    default: return NullNode();
+  }
+}
+
+ASTNode funcATan(ASTNode radian){
+  switch(radian.type){
+    case ASTType.AST_INT: return DoubleNode()..doubleVal = atan(radian.intVal);
+    case ASTType.AST_DOUBLE: return DoubleNode()..doubleVal = atan(radian.doubleVal);
+    default: return NullNode();
+  }
```

Comment:
For each of these functions:
- They should be following the `ASTNode methodName(Runtime runtime, ASTNode self, List<ASTNode> args) ` format.
- Cascade notation returns the original object.
- Can be simplified by using the `doubleVal`
- Expect all arguments rather than returning null